src/providers/LocalVideoProvider/index.tsx

```diff
@@ -32,25 +33,50 @@ const LocalVideoProvider: React.FC = ({ children }) => {
       setIsVideoEnabled(true);
     }
 
+    const observer: AudioVideoObserver = {
+      videoAvailabilityDidChange: (availability) => {
+        if (!availability.canStartLocalVideo) {
+          setIsReachVideoLimit(true);
+        } else {
+          setIsReachVideoLimit(false);
+        }
+        console.log(
+          'video availability changed: canStartLocalVideo ',
+          availability.canStartLocalVideo
+        );
+      },
+    };
+    audioVideo.addObserver(observer);
+
     return () => {
       setIsVideoEnabled(false);
+      audioVideo.removeObserver(observer);
     };
   }, [audioVideo]);
 
+  useEffect(() => {
+    if (isReachVideoLimit) {
+      console.warn('Reach the number of maximum active videos');
+    }
+  }, [isReachVideoLimit]);
+
   const toggleVideo = useCallback(async (): Promise<void> => {
     if (isVideoEnabled || !meetingManager.selectedVideoInputTransformDevice) {
```

Comment:
Wonder if we should add a console.warn if isVideoEnabled is false but meetingManager.selectedVideoInputTransformDevice or there is no input device chosen.

Reply:
Add a warning log if customer want to turn on the video while there is no input video device chosen

Reply:
I dont think you need the first check `!isVideoEnabled`

Reply:
Good catch!